app/furniture/marketplace/guest.rb

```diff
@@ -0,0 +1,16 @@
+class Marketplace
+  class Guest < ::Guest
+    attr_writer :shopper
+    def shopper
+      @shopper ||= Shopper.find_by(id: session[:guest_shopper_id])
+    end
+
+    def find_or_create_shopper
+      @shopper ||= Shopper.find_or_create_by(id: session[:guest_shopper_id] ||= SecureRandom.uuid)
+    end
+
+    private def guest_shopper_id
```

Comment:
Is this method used anywhere? I can't find any references to it.

Reply:
Nope, I defined it but didn't use it...